src/aker_metrics.c

```diff
@@ -19,161 +19,188 @@
+
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+void stringify_metrics(int flag)
 {
 	char str[512];
 	aker_metrics_t * tmp = NULL;
 	tmp = g_metrics;
 
-	if(g_metrics != NULL && flag == 1)
+	if(g_metrics != NULL)
 	{
-		snprintf(str, 512, "%s%d%s%d%s%d%s%d%s%ld%s%d%s%s",
-						"DeviceBlockCount,", tmp->device_block_count,
-                                                  ",WindowTransistionCount,", tmp->windows_transistion_count,
-                                                    ",ScheduleSetCount,", tmp->schedule_set_count,
-                                                    ",MD5ErrorCount,", tmp->md5_error_count,
-                                                    ",ProcessStartTime,", tmp->process_start_time,
-                                                    ",ScheduleEnabled,", tmp->schedule_enabled,
-                                                    ",TimeZone,", tmp->timezone);
+		snprintf(str, 512, "DeviceBlockCount,%d,"
+                   "WindowTransistionCount,%d,"
+                   "ScheduleSetCount,%d,"
+                   "MD5ErrorCount,%d,"
+                   "ProcessStartTime,%ld,"
+                   "ScheduleEnabled,%d,"
+                   "TimeZone,%s,"
+                   "TimeZoneOffset,%+d",
+
+                   tmp->device_block_count,
+                   tmp->window_trans_count,
+                   tmp->schedule_set_count,
+                   tmp->md5_err_count,
+                   tmp->process_start_time,
+                   tmp->schedule_enabled,
+                   tmp->timezone,
+                   tmp->timezone_offset);
+
 
 		debug_info("The stringified valued is (%s)\n", str);
 
-#if defined(ENABLE_FEATURE_TELEMETRY2_0)
-		t2_event_s("akermetric", str);
-		debug_info("akermetric t2 event triggered\n");
-#endif
+		if(flag)
+		{
+
+	#if defined(ENABLE_FEATURE_TELEMETRY2_0)
+			t2_event_s("akermetrics", str);
+			debug_info("akermetrics t2 event triggered\n");
+	#endif
+		}
 
 	}
 	else
 	{
-		if(flag == 0)
-		{
-			debug_info("The current stringified value is (DeviceBlockCount, %d, WindowTransistionCount, %d, ScheduleSetCount, %d, MD5ErrorCount, %d, ProcessStartTime, %ld, ScheduleEnabled, %d, TimeZone, %s)\n", tmp->device_block_count, tmp->windows_transistion_count, tmp->schedule_set_count, tmp->md5_error_count, tmp->process_start_time, tmp->schedule_enabled, tmp->timezone);
-
-		}
-		else
-		{
-			debug_error("The g_metrics is NULL\n");
-		}
-		return 0;
+		debug_error("The g_metrics is NULL\n");
 	}
-
-	return 1;
 }
 
-int get_blocked_mac_count(char* blocked)
+/* See aker_metrics.h for details. */
+int get_blocked_mac_count(const char* blocked)
 {
 	int count = 0;
-	char* mac =  strndup(blocked, strlen(blocked));
 
-	debug_print("The mac obtained were %s\n", mac);
+	debug_info("The mac obtained were %s\n", blocked);
+
 	// Returns first mac with delimiter as " "
-	char* mac_token = strtok(mac, " ");
+	char* mac_token = strtok((char *)blocked, " ");
```

Comment:
Don't use `strtok()` for this code as it is not needed for this code.  You are iterating across the entire string looking for spaces.  strtok() may change your string.  It's better to loop across the bytes & simply count the `\s` characters and add one.

See the loop I added yesterday.